File: public/docs/_examples/architecture/ts/app/sales-tax.service.ts
// #docregion
import { Inject, Injectable }     from '@angular/core';

import { TaxRateService } from './tax-rate.service';

// #docregion class
@Injectable()
export class SalesTaxService {
  constructor(private rateService: TaxRateService) { }
  getVAT(value:string | number){
    let amount:number;
    if (typeof value === "string"){
      amount = parseFloat(value);
    } else {
      amount = value;
    }
    return (amount || 0) * this.rateService.getRate('VAT');
  }
}
// #enddocregion class

File: public/docs/_examples/architecture/dart/lib/sales_tax_service.dart
import 'package:angular2/core.dart';

import 'tax_rate_service.dart';

@Injectable()
class SalesTaxService {
  TaxRateService rateService;

  SalesTaxService(this.rateService);

  num getVAT(dynamic /* String | num */ value) =>
      rateService.getRate('VAT') *
      (value is num ? value : num.parse(value, (_) => 0));
}
